code/datums/components/rust.dm

```diff
@@ -0,0 +1,83 @@
+/datum/component/rust
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+	var/atom/parent_atom
+	var/mutable_appearance/rust_overlay
+	var/static/list/typecache_of_valid_types = list(
```

Comment:
You shouldn't have this.

Components should be able to be applied to everything that *technically* works, not just what the designer wanted.

For example, you'd check at the very least `isatom`, since it isn't *possible* to apply rust to just any datum. Work from there.

Reply:
Technically I already have the check that ensures it works, it just checks for a "rust" icon state to apply as an overlay